tools/cdk-release/test/changelog.test.ts

```diff
@@ -1,14 +1,130 @@
+      { ...args, changelogExperimentalChanges: 'include' },
+      currentVersion, newVersion, commits, packages);
+
+    expect(Object.keys(changelogResult)).toEqual(['CHANGELOG.md']);
+    expect(changelogResult['CHANGELOG.md'].trim()).toBe(
+      `## [1.24.0](https://github.com/aws/aws-cdk/compare/v1.23.0...v1.24.0)
+
+### Features
+
+* **aws-experimental:** new experimental feat
+* **aws-stable:** new stable feat`);
+  });
+
+  test('if changelogExperimentalChanges is "strip", excludes experimental changes', async () => {
+    const changelogResult = await writeChangelogs(
+      { ...args, changelogExperimentalChanges: 'strip' },
+      currentVersion, newVersion, commits, packages);
+
+    expect(Object.keys(changelogResult)).toEqual(['CHANGELOG.md']);
+    expect(changelogResult['CHANGELOG.md'].trim()).toBe(
+      `## [1.24.0](https://github.com/aws/aws-cdk/compare/v1.23.0...v1.24.0)
+
+### Features
+
+* **aws-stable:** new stable feat`);
+  });
+
+  test('if changelogExperimentalChanges is "separate", throws if alpha versions are not present', async () => {
+    await expect(writeChangelogs(
+      { ...args, changelogExperimentalChanges: 'separate' },
+      { stableVersion: '1.23.0' },
+      { stableVersion: '1.24.0' },
+      commits, packages))
+      .rejects
+      .toThrow(/without alpha package versions/);
+
+    await expect(writeChangelogs(
+      { ...args, changelogExperimentalChanges: 'separate' },
+      { stableVersion: '1.23.0', alphaVersion: '1.23.0-alpha.0' },
+      { stableVersion: '1.24.0' },
+      commits, packages))
+      .rejects
+      .toThrow(/without alpha package versions/);
+
+    await expect(writeChangelogs(
+      { ...args, changelogExperimentalChanges: 'separate' },
+      { stableVersion: '1.23.0' },
+      { stableVersion: '1.24.0', alphaVersion: '1.24.0-alpha.0' },
+      commits, packages))
+      .rejects
+      .toThrow(/without alpha package versions/);
+  });
+
+  test('if changelogExperimentalChanges is "separate", excludes experimental changes and writes to separate changelog', async () => {
+    currentVersion.alphaVersion = '1.23.0-alpha.0';
+    newVersion.alphaVersion = '1.24.0-alpha.0';
+    const changelogResult = await writeChangelogs(
+      { ...args, changelogExperimentalChanges: 'separate' },
+      currentVersion, newVersion, commits, packages);
+
+    expect(Object.keys(changelogResult)).toEqual(expect.arrayContaining(['CHANGELOG.md', 'aws-experimental/CHANGELOG.md']));
+    expect(changelogResult['CHANGELOG.md'].trim()).toBe(
+      `## [1.24.0](https://github.com/aws/aws-cdk/compare/v1.23.0...v1.24.0)
+
+### Features
+
+* **aws-stable:** new stable feat`);
+    expect(changelogResult['aws-experimental/CHANGELOG.md'].trim()).toBe(
+      `## [1.24.0-alpha.0](https://github.com/aws/aws-cdk/compare/v1.23.0-alpha.0...v1.24.0-alpha.0)
+
+### Features
+
+* **aws-experimental:** new experimental feat`);
+  });
+});
 
+describe('changelog', () => {
```

Comment:
Any chance we can refactor these to use `writeChangelog()` instead, and thus make `changelog()` private?

Reply:
We could, but I like the idea of testing this method directly, and going more into depth on it. We're going to add more logic to it (and the current tests could use some work even as-is). I'd rather unit test the changelog transforms and generation itself directly where possible.